Adding Viewport Workflow

LB-JaroslawGaweda opened the pull request:
Signed-off-by: Jarosław Gawęda <99716227+LB-JaroslawGaweda@users.noreply.github.com>
## Change summary
Adding the Viewport workflow file.

hultonha commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,74 @@
+​
+## Common Issues to Watch For
+​
+Test guidance will sometimes note specific issues to watch for. The common issues below should be watched for through all testing, even if unrelated to the current workflow being tested.
+1. Console log errors/warnings/spam
```
nit: Please leave line breaks around lists

FuzzyCarterAWS replied:
```suggestion
Test guidance will sometimes note specific issues to watch for. The common issues below should be watched for through all testing, even if unrelated to the current workflow being tested.

1. Console log errors/warnings/spam
```

hultonha commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,74 @@
+1. Console log errors/warnings/spam
+2. Asserts
+​
+### Platforms:​
+- Windows
```
nit: Please add line breaks around headings (repeated)

FuzzyCarterAWS replied:
```suggestion
### Platforms:​

- Windows
```

hultonha commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,74 @@
+| Workflow                                                               | Requests																																																																						   | Things to Watch For                                                                                                                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------- ...
+| Navigate level with camera controls									 | <ol><li>Select the Viewport.</li><li>Navigate the camera with W, A, S, D, Q, E. Hold shift for fast move.</li><li>Right-click and drag to rotate the camera.</li><li>Press MMB and drag to pan up and down.</li><li>Scroll up and down using the MMB, to zoom in and out.</li><li>ALT+LMB to orbit.</li></ol>| <ul><li>Correct camera behaviour.</li></ul>				 ...
+| Be this camera option													 | <ol><li>Select the camera entity or create one.</li><li>In the Entity Inspector, press the "Be this camera" button.</li><li>Move the camera around.</li><li>Press the Return to default editor camera button.</li></ol>																		   | <ul><li>Correct camera behaviour.</li></ul>																																  |
```
suggestion: It might also be worth mentioning 'Match view' which would be good to sanity check (it moves the camera to the current viewport transform).

FuzzyCarterAWS replied:
```suggestion
| Be this camera option													 | <ol><li>Select the camera entity or create one.</li><li>In the Entity Inspector, press the `Be this camera` button.</li><li>Move the camera around.</li><li>Press the Return to default editor camera button.</li></ol>																		   | <ul><li>Correct camera behaviour.</li></ul>																																  |
```

hultonha commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,74 @@
+| Viewport camera selector												 | <ol><li>Open the Viewport Camera Selector (Tools > Viewport > Viewport Camera Selector).</li><li>Right-Click in the Viewport and select "Create camera entity from view".</li><li>Navigate to a different location in the Viewport and create another camera.</li><li>Navigate through the available cameras in the Viewport Camera Selector while monitoring the View ...
+---
+
+### Workflow 2: Viewport options  
```
It would be good to have an explicit section about the viewport title dialog (top right of viewport) to sanity check those options work correctly (e.g. FOV and show/hide helpers/icons etc...)

hultonha reviewed (approved):
Overall this looks like a great start, I'd love for @FiniteStateGit to have a read through this and offer some suggestions given their familiarity with the viewport having documented it recently.

It would be good to include a section about view bookmarks (Ctrl-Fn<1-12> keys to set and Shift-Fn<1-12> to restore).

It would also be good to have a performance/stress test section (verify having lots of entities can be quickly box selected without causing large performance slow downs). Also testing certain settings persist would be good too.

Tagging @HogJonny-AMZN and @benblack75 for visibility too as it'd be great to have them review this briefly. Thanks!

FuzzyCarterAWS commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,74 @@
+- Windows
+- Linux​
+​
+### Documentation pages:​
+* [Working with the 3D Viewport](https://www.o3de.org/docs/user-guide/editor/Viewport/)
+* [Reference Spaces](https://www.o3de.org/docs/user-guide/editor/Viewport/reference-spaces/)
+* [Group Selection](https://www.o3de.org/docs/user-guide/editor/Viewport/group-selection/)
+* [Advanced 3D Viewport Operations](https://www.o3de.org/docs/user-guide/editor/Viewport/advanced-operations/)​
```
```suggestion
### Documentation pages:​

* [Working with the 3D Viewport](https://www.o3de.org/docs/user-guide/editor/Viewport/)
* [Reference Spaces](https://www.o3de.org/docs/user-guide/editor/Viewport/reference-spaces/)
* [Group Selection](https://www.o3de.org/docs/user-guide/editor/Viewport/group-selection/)
* [Advanced 3D Viewport Operations](https://www.o3de.org/docs/user-guide/editor/Viewport/advanced-operations/)​
```

FuzzyCarterAWS commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,74 @@
+
+| Workflow                                                               | Requests																																																																						   | Things to Watch For                                                                                                                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------- ...
+| Navigate level with camera controls									 | <ol><li>Select the Viewport.</li><li>Navigate the camera with W, A, S, D, Q, E. Hold shift for fast move.</li><li>Right-click and drag to rotate the camera.</li><li>Press MMB and drag to pan up and down.</li><li>Scroll up and down using the MMB, to zoom in and out.</li><li>ALT+LMB to orbit.</li></ol>| <ul><li>Correct camera behaviour.</li></ul>				 ...
```
If we use code wraps around button presses it'll help highlight the actions within the workflow while skimming through the doc.

Let's call out an additional thing or two to watch out for here aside from just "correct camera behaviour". 

Something along the lines of:
Camera responds as expected to the used control actions.
Camera controls do not invert while switching between operations.

```suggestion
| Navigate level with camera controls									 | <ol><li>Select the Viewport.</li><li>Navigate the camera with `W`, `A`, `S`, `D`, `Q`, `E`. Hold shift for fast `move.</li><li>`Right-click and drag` to rotate the camera.</li><li>Press `MMB and drag` to pan up and down.</li><li>Scroll up and down using the `MMB`, to zoom in and out.</li><li>`ALT+LMB` to orbit.</li></ol>| <ul><li>Correct camera behaviour.</li></ul>																													  |
```

LB-JaroslawGaweda commented:
@hultonha Thanks for the input. I'll wait if there are other suggestions from the reviewers and make the changes.

hultonha commented:
> @hultonha Thanks for the input. I'll wait if there are other suggestions from the reviewers and make the changes.

Thanks @LB-JaroslawGaweda 👍

FuzzyCarterAWS commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,77 @@
+
+| Workflow                                                               | Requests																																																																						   | Things to Watch For                                                                                                                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------- ...
+| Navigate level with camera controls									 | <ol><li>Select the Viewport.</li><li>Navigate the camera with `W`, `A`, `S`, `D`, `Q`, `E`. Hold shift for fast `move.</li><li>`Right-click and drag` to rotate the camera.</li><li>Press `MMB and drag` to pan up and down.</li><li>Scroll up and down using the `MMB`, to zoom in and out.</li><li>`ALT+LMB` to orbit.</li></ol>| <ul><li>Correct camera beh ...
```
Looks like there's a missing opening `\``

LB-JaroslawGaweda replied:
Fixed the rogue `.  I will update the rest of the workflow with code wraps around button presses.

hultonha commented:
@benblack75 are you happy to remove you 'Request Changes' now? Thanks!

hultonha commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,77 @@
+
+| Workflow                                                               | Requests																																																																						   | Things to Watch For                                                                                                                                                          |
+|------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------- ...
+| Navigate level with camera controls									 | <ol><li>Select the Viewport.</li><li>Navigate the camera with `W`, `A`, `S`, `D`, `Q`, `E`. Hold `shift` for fast move.</li><li>`Right-click and drag` to rotate the camera.</li><li>Press `MMB and drag` to pan up and down.</li><li>Scroll up and down using the `MMB`, to zoom in and out.</li><li>`ALT+LMB` to orbit.</li></ol>| <ul><li>Correct camera be ...
```
suggestion: It would be good to mention using Pan/Scroll while also holding Alt too (as these behaviors also work while orbiting the camera). It might also be worth mentioning testing clicking in and out of the viewport while moving the camera, as in the past we have had a few subtle issues with focus causing problems when clicking in and out of the viewport.

hultonha reviewed (approved):
Change looks good, the one other area I think is missing (as well as maybe mentioning a performance stress test) is the Component Mode Switcher (tagging @amzn-ahmadkrm for visibility). It would be great to have a section on this to verify going in and out of Component Mode works correctly. This is related to the above and the existing workflow but I'd also mention testing undo/redo as this is something that has been known to break or cause problems and is critical for users.

LB-JaroslawGaweda commented:
I have added the mentioned workflow tasks from the comments.

@hultonha About the Match view, I could not find any information about this in documentation, I'm not quite sure if this is it, but I have added the `Find the selected entity in the Viewport` task.

hultonha commented:
> I have added the mentioned workflow tasks from the comments.
> 
> @hultonha About the Match view, I could not find any information about this in documentation, I'm not quite sure if this is it, but I have added the `Find the selected entity in the Viewport` task.

Hi @LB-JaroslawGaweda, hmm it unfortunately might not be documented but I was simply referring to this button

![image](https://user-images.githubusercontent.com/82228511/227542738-a6b2bcc4-f5a7-423b-8e10-5605d525ddae.png)

It would be good to check this behaves as expected (simply a bullet point to say 'Test match viewport' should be sufficient). Thanks!

FuzzyCarterAWS commented on testing-guidance/workflow-tests/Viewport/viewport-workflow-tests.md:
```diff
@@ -0,0 +1,87 @@
+| Matching entity transforms											 | <ol><li>Select an entity.</li><li>Press and hold `Ctrl` and press the middle mouse button as you hover over a target entity. After the target is selected, the current selection transform matches that of the target.</li><li>Do this with location, rotation, or scale.</li></ol>			   | <ul><li>Correct matching location, rotation, and scale values.</li></ul>		 ...
+| Ditto a group selection											     | <ol><li>Select a group of entities.</li><li>Press and hold `Ctrl` and press the `middle mouse button` as you hover over a target entity to match the group's transform data to the target's data.</li><li>Do this with location, rotation, or scale.</li></ol>									   | <ul><li>Correct matching location, rotation, and scale values.</li></ul>																 ...
+| Ditto a group selection to local space								 | <ol><li>Select a group of entities.</li><li>Press and hold `Ctrl+Alt` and press the `middle mouse button` as you hover over a target entity. This sets the local space of each entity in the selected group to the target entity that you specified.</li></ol>								       | <ul><li>Correct matching location, rotation, and scale values.</li></ul>					 ...
+---
```
NIT: Blank like at end of file.

FuzzyCarterAWS reviewed (approved):
This looks great

LB-JaroslawGaweda commented:
@hultonha I have added the Match Viewport workflow task and did minor fixes. If there are no more suggestions, it is ready to be merged.

hultonha commented:
> @hultonha I have added the Match Viewport workflow task and did minor fixes. If there are no more suggestions, it is ready to be merged.

Perfect! Thanks @LB-JaroslawGaweda, I'll merge now 👍

LB-JaroslawGaweda commented:
Confirming that the pull request has been merged.
